stacks-signer/src/v0/signer.rs

```diff
@@ -1136,9 +1160,45 @@ impl Signer {
         }
     }
 
-    /// Calculate the tenure extend timestamp based on the tenure start and already consumed idle time
-    fn calculate_tenure_extend_timestamp(&self) -> u64 {
-        // TODO: udpate this to grab the idle time consumed against the tenure start time
-        get_epoch_time_secs()
+    /// Calculate the tenure extend timestamp based on the tenure start and already consumed idle time.
+    fn calculate_tenure_extend_timestamp(
+        &self,
+        stacks_client: &StacksClient,
+        sortition_state: &mut Option<SortitionsView>,
+    ) -> u64 {
+        if sortition_state.is_none() {
+            *sortition_state =
+                SortitionsView::fetch_view(self.proposal_config.clone(), stacks_client)
+                    .inspect_err(|e| {
+                        warn!(
+                            "{self}: Failed to update sortition view: {e:?}";
+                        )
+                    })
+                    .ok();
+        }
+        if let Some(sortition_state) = sortition_state {
+            let tenure_process_time = self
+                .signer_db
+                .get_globally_accepted_blocks(&sortition_state.cur_sortition.consensus_hash)
+                .unwrap_or_default()
+                .iter()
+                .map(|block| {
+                    if let Some(processed_time) = block.processed_time {
+                        processed_time.saturating_sub(block.proposed_time)
+                    } else {
+                        0
+                    }
+                })
+                .sum::<u64>();
+
+            sortition_state
+                .cur_sortition
+                .burn_header_timestamp
```

Comment:
This won't work i don't think...cause the burn header timestamp could be super duper old. Like this could be the second, third,...,Nth tenure extend. I will need to track the current tenure timestamp start better I think.

Reply:
We can relatively easily check if a block has a TenureChange payload in it (doesn't matter if it's for an extend or new block), and then use that as our "start" for the timer

Reply:
yeah I will add this as a flag for the blocks in the database for easier retrieval of the last globally accepted tenure change block.

Reply:
Yep that makes sense to me.